src/controllers/ChatsController.ts

```diff
@@ -28,11 +29,38 @@ export class ChatsController {
     }
   }
 
+  checkAndSetParameterInRoute() {
+    const parameter = router.getParameter();
+    if (parameter === null) {
+      store.set('currentChatId', parameter);
+      return parameter;
+    }
+    const validatedId = this.validateChatId(Number(parameter));
+    store.set('currentChatId', validatedId);
+    return validatedId;
+  }
+
+  toGoToChatById(id: number | string | null) {
+    if (id === null) {
+      store.set('currentChatId', null);
+      router.go(Routes.Messenger, null);
+    } else {
+      store.set('currentChatId', id);
+      router.go(Routes.Messenger, Number(id));
+    }
+  }
```

Comment:
Кажется тут if ни на что не влияет
```suggestion
  goToChatById(id: number | string | null) {
      store.set('currentChatId', id);
      router.go(Routes.Messenger, id);
  }
```

Reply:
Number приводит null к 0. А мне для корректной работы компонентов надо что бы в currentChatId при отсутствии чата ставился именно null. Возможно это мой изначальный просчет при проектировании, но теперь проще оставить здесь проверку, чем переделывать компоненты.